internal/uvm/mem_allocator.go

```diff
@@ -0,0 +1,182 @@
+package uvm
+
+import (
+	"github.com/pkg/errors"
+)
+
+const (
```

Comment:
Should these be their own type?

Reply:
+ any reason these are exported?

Reply:
type yes, exported - not with this version, but I was definitely going the route of making a different package etc.